docs/repo/property-pages/back-end-architecture.md

```diff
@@ -34,6 +34,8 @@ The Project Query API treats its use of brokered services as an implementation d
 
 Internally, the Project Query API handles serializing the query, sending it across the network, deserializing it, and coordinating the actions of an extensible set of data providers. These data providers are responsible for filling in the requested properties on various entities, creating child entities from their parents, and handling various "actions" that update state in some way. For example, one provider may be responsible for creating project entities representing all the projects in the solution, while a different provider is responsible for filling in the project's name, path, GUID, language, etc. There may be multiple providers that handle the same relationship from a parent to its children or the same action. Among other things, the Project Query API implementation glues together the results of all of these providers to create the final query result.
 
+Please note that internally, setting either a property's evaluated or unevaluated value to null will call `IProject.DeleteAsync` to delete the property.
```

Comment:
I think this information should go in the comments for the relevant types in the CPS system. This is, fundamentally, a change in the semantics of the Project Query API itself (though we currently have the only implementation of that part of the API).

Reply:
I will make sure to document this. Taking on #6921